### DISCUSSION TITLE
Problemas de envio

### DISCUSSION THREAD

**Discussion by leomare12** [rramosp/ai4eng.v1#231] [02 - PYTHON]
Hola profesor, 

para el lab 2.01 task 4 cree el código y al evaluarlo para confirmar manualmente para diferentes valores me da correcto ...

perfecto(6), perfecto(17), perfecto(28), perfecto(397), perfecto(496), perfecto(8128), perfecto(9385)

OUT: (True, False, True, False, True, True, False)

pero al enviar mi código aparece lo siguiente:

error: ERROR EXECUTING YOUR CODE
---------------------------------
Traceback (most recent call last):
  File "calling_script.py", line 24, in <module>
    r = eval("efjgdclb.{}(source_functions, source_variables, '[leonardo.marin@udea.edu.co](mailto:leonardo.marin@udea.edu.co)')".format(grader_function_name), namespace)
  File "<string>", line 1, in <module>
  File "/content/efjgdclb.py", line 28, in grader_04
    answer = st_perfecto(n)
  File "<string>", line 2, in perfecto
NameError: name 'math' is not defined
 (set session.debug=True for tracebacks)

... correctamente logueado:
logging in as [leonardo.marin@udea.edu.co](mailto:leonardo.marin@udea.edu.co)... please wait
using course session ai4eng.v1.udea.r2::2023.1
success!! you are logged in

y la librería math la importo y ejecuto.

¿Qué debo hacer? ya llevo varios intentos y es con el único que no he podido.

Quedo atento a su respuesta. Gracias.

**Comment by rramosp** [comment]
comparte tu código ... probablemente tengas que importar math **dentro** de
tu función ... el corrector solo ve lo que hay dentro de las funciones de
los ejercicios, no lo que hay en el resto del notebook

On Mon, Feb 27, 2023 at 10:31 PM leomare12 ***@***.***> wrote:

> Hola profesor,
>
> para el lab 2.01 task 4 cree el código y al evaluarlo para confirmar
> manualmente para diferentes valores me da correcto ...
>
> perfecto(6), perfecto(17), perfecto(28), perfecto(397), perfecto(496),
> perfecto(8128), perfecto(9385)
>
> OUT: (True, False, True, False, True, True, False)
>
> pero al enviar mi código aparece lo siguiente:
> error: ERROR EXECUTING YOUR CODE
>
> Traceback (most recent call last):
> File "calling_script.py", line 24, in
> r = eval("efjgdclb.{}(source_functions, source_variables, '
> ***@***.***')".format(grader_function_name), namespace)
> File "", line 1, in
> File "/content/efjgdclb.py", line 28, in grader_04
> answer = st_perfecto(n)
> File "", line 2, in perfecto
> NameError: name 'math' is not defined
> (set session.debug=True for tracebacks)
>
> ... correctamente logueado:
> logging in as ***@***.*** please wait
> using course session ai4eng.v1.udea.r2::2023.1
> success!! you are logged in
>
> y la librería math la importo y ejecuto.
>
> ¿Qué debo hacer? ya llevo varios intentos y es con el único que no he
> podido.
>
> Quedo atento a su respuesta. Gracias.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rramosp/ai4eng.v1/discussions/231>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADJRZII77UOXXW4GXXLMJ6TWZVWP7ANCNFSM6AAAAAAVKGB6DI>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

-- 
*Aviso legal:* El contenido de este mensaje y los archivos adjuntos son 
confidenciales y de uso exclusivo de la Universidad Nacional de Colombia. 
Se encuentran dirigidos sólo para el uso del destinatario al cual van 
enviados. La reproducción, lectura y/o copia se encuentran prohibidas a 
cualquier persona diferente a este y puede ser ilegal. Si usted lo ha 
recibido por error, infórmenos y elimínelo de su correo. Los Datos 
Personales serán tratados conforme a la Ley 1581 de 2012 y a nuestra 
Política de Datos Personales que podrá consultar en la página web 
www.unal.edu.co <http://www.unal.edu.co/>.* *Las opiniones, informaciones, 
conclusiones y cualquier otro tipo de dato contenido en este correo 
electrónico, no relacionados con la actividad de la Universidad Nacional de 
Colombia, se entenderá como personales y de ninguna manera son avaladas por 
la Universidad.

**Comment by leomare12** [comment]
Hola Profesor.... Lo tengo en la primera línea de mi código. El import math también lo usé en el 2.01 task 2 sin problema 

import math
def perfecto(n):
  a = math.floor(n/2)
  b = 0
  for i in range (1,a + 1):
    if n%i == 0:
      b += i
    else:
      continue
  if n == b:
    return True
  else:
    return False

Quedo atento. Gracias

**Reply by Felipe-RA** [reply]
Hola @leomare12 

Debes ponerlo **dentro** de tu función.

Tu función es lo que se envía al calificador, como está por fuera el `import math` no se envía

**Reply by leomare12** [reply]
Solución correcta, Muchas gracias.
